build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -172,6 +172,30 @@ service ActionCache {
   }
 }
 
+// The repository cache API is used to download and cache external repositories.
+// Once downloaded, the repository content can be fetched from the
+// [ContentAddressableStorage][build.bazel.remote.execution.v2.ContentAddressableStorage]
+// by digest.
+service RepositoryCache {
+  // Download and cache a single file.
+  //
+  // Errors:
+  // * `INVALID_ARGUMENT`: One or more request fields are invalid.
+  // * `NOT_FOUND`: The cache could not find content for the given request.
+  // * `RESOURCE_EXHAUSTED`: There is insufficient CAS capacity to store the
+  //   downloaded content.
+  // * `FAILED_PRECONDITION`: The cache found content at the given URLs that it
+  //   believes is valid content, but it did not match the content integrity
+  //   checksum.
+  //
+  // Repository caches may, but are not required to, verify content integrity.
```

Comment:
See my comment below for an alternate requirement (if any, at least one must match). I don't feel strongly about *those* semantics - another option is for the server to say which integrity formats they matched explicitly - but I do think clients need something stronger than "may" here. If a client provides an integrity string but cannot tell if it matched or was ignored, what can they do with that information?

Reply:
I can't think of any cases in which the client would want to change its behavior based on the presence of a response field with a content checksum in it.

As I understand the security model of a distributed build cluster, either:
* The client does not trust the repository cache, and thus the client must re-validate downloaded files regardless of what the cache says. A response field with the cache's opinion on integrity would be ignored.
* The client does trust the repository cache, and does not re-validate downloaded files. A response field with the cache's opinion on integrity would be ignored.

Either way the behavior is exclusively in the domain of the client, controlled by client options like Bazel's `--noremote_verify_downloads`.

Reply:
If a client wants to ensure that either it or the server validated the files, it presumably needs to know whether the server did. So this is the "client trusts the server" case.